### DISCUSSION TITLE
How to disable Rack Awareness (and kafka-init of Kafka-brokers respectively)?

### DISCUSSION THREAD

**Discussion by herrfeder** [strimzi/strimzi-kafka-operator#10410] [Q&A] [answered]
Hi,

because of special Network Requirements (using istio-cni and therefore we don't have network connectivity in init-containers at all) the kafka-init Container of the Kafka-Brokers isn't capable of doing the GET Kubernetes-API-Request for Node to generate the Rack ID.

Because of other reasons, we are doing Kafka-Broker Pod AntiAffinity on our own and therefore we don't need the kafka-init Container to establish Rack Awareness.

I reviewed the Strimzi Code and saw that there are several `rack != null` checks: https://github.com/strimzi/strimzi-kafka-operator/blob/ff128d8fbf9e964671c6b6367e571c69fa172ce6/cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java#L1495

According to the Strimzi Documentation, the rack configuration is configured accordingly: https://github.com/strimzi/strimzi-kafka-operator/blob/main/documentation/api/io.strimzi.api.kafka.model.common.Rack.adoc

-> but I didn't found a way to influence the resulting behaviour at all or found any hint in the source-code which CRD-Element I have to trigger to disable this init container.

I checked also the TopicOperator for a way to insert a `--disable-rack-aware` but it seems, the PodArgs are hardcoded.

So in the end, is there a supported way to disable the rack awareness feature and therefore the kafka-init Container?

**Comment by scholzj** [comment, marked as answer]
You can disable Rack awareness by removing it from the Kafka CR - you had to enable it there in the first place. However, you should keep in mind that:
* Strimzi does not support Istio, so not sure if this is the only thing that would not work
* The rack awareness is not about setting affinity rules. It is about the broker knowing in which rack it is in in order to place the partition replicas. So by setting affinity manually, you do not replace it in any way.

**Reply by herrfeder** [reply]
Thank you for the answer. Yes we had serious issues to get Strimzi working at all with Istio and is incooperating some workarounds.
And thank you for further explaination regarding the purpose of the rack awareness and it isn't only Deployment Replicas, but also partition replicas, that's a completely different story when it comes to the decision to disable it.

Thank you again. That means I have to add another istio workaround to handle it :)
